src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -2220,10 +2242,18 @@ ColumnPtr executeStringInteger(const ColumnsWithTypeAndName & arguments, const A
             return executeTupleNumberOperator(arguments, result_type, input_rows_count, function_builder);
         }
 
+        /// Result can be nullable enven if input arguments are not nullable
+        if (is_divide_or_null || is_modulo_or_null)
+        {
+            NullMap res_null_map(input_rows_count, 0);
+            auto res = executeImpl2(arguments, result_type, input_rows_count, nullptr, &res_null_map);
+            return wrapInNullable(res, arguments, result_type, input_rows_count, &res_null_map);
+        }
```

Comment:
It seems it's better to move this change to `executeImpl2()` - because `executeImpl2()` already calculates `right_nullmap` which a similar thing and it's better to keep similar things near each others.

Also `res_null_map` seems very similar to `right_null_map` - so it seems `executeImpl2()` could copy `right_null_map` to `res_null_map` so after that `BinaryOperationImpl<A, B, ModuloOrNullImpl<A, B>>::process()` would have to deal with only one null map.

Reply:
OK. I am on a vacation, will do it later.

Reply:
Work is more than I think.

Reply:
@vitlibar Pls review if you have time.